.changeset/wise-boxes-develop.md

```diff
@@ -0,0 +1,42 @@
+---
+'astro': minor
+---
+
+Exports a new `getActionPath()` helper from `astro:actions`
+
+The `getActionPath()` utility accepts an action and returns a URL path so you can execute an action call as a `fetch()` operation directly. This allows you to provide details such as custom headers when you call your action.
+
+This example shows how to call a defined `like` action passing the `Authorization` header and the [`keepalive`](https://developer.mozilla.org/en-US/docs/Web/API/Request/keepalive) option:
+
+```astro
+<script>
+// src/components/my-component.astro
+import { actions, getActionPath } from 'astro:actions'
+
+await fetch(getActionPath(actions.like), {
+  method: 'POST',
+  headers: {
+    'Content-Type': 'application/json',
+    Authorization: 'Bearer YOUR_TOKEN'
+  },
+  body: JSON.stringify({ id: 'YOUR_ID' }),
+  keepalive: true
+})
```

Comment:
Thanks for adding this! I think it's definitely worth noting that **actions do not return plain JSON.** They return values that are parsable by the `devalue` library. I suggest using the `deserializeActionResult()` utility from `astro:actions` for this.

Reply:
Ah good call thank you!

Reply:
Thanks @bholmesdev -- is *showing* using `deserializeActionResult()` in this context concise? I realized we only have a short API reference sentence for this (no example usage), and no example using this anywhere in docs. (So, I'm not sure if this can be shown quickly or not.)

This would be a great addition to the actual docs (where I don't really care if it's concise or not), and if it's easy to do here, would also be a great place to put it!